pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +270,23 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns list of users encoded in spiffe format from the local trust domain
+// and its aliases.
+// For example, if the local trust domain is "cluster.local", and its aliases are "td1", "td2".
+// For a user "bar" in namespace "foo", getPrincipalsIncludingAliases returns
+// cluster.local/ns/foo/sa/bar, td1/ns/foo/sa/bar, td2/ns/foo/sa/bar
+func getPrincipalsIncludingAliases(trustDomainAliases []string, principals []string) []string {
+	usersWithAliases := principals
+	for _, tdAlias := range trustDomainAliases {
+		for _, principal := range principals {
+			identityParts := strings.Split(principal, "/")
```

Comment:
Do we have these parse/construct principals utils somewhere? May be useful to have that util functions.

Reply:
Not sure, but since it's simple enough, we can create one? I added a util function for parsing the constructing a new identity.